core/mempool/src/lib.rs

```diff
@@ -45,7 +48,7 @@ impl Pool {
     pub fn new(signer: Arc<BlockSigner>, storage: Arc<RwLock<ShardChainStorage>>, trie: Arc<Trie>) -> Self {
         Pool {
             signer,
-            transactions: RwLock::new(HashSet::new()),
+            transactions: RwLock::new(HashMap::new()),
```

Comment:
`Default::default()` for `transactions` and `receipts`.

Reply:
Fixed